vault/core.go

```diff
@@ -896,8 +897,18 @@ func NewCore(conf *CoreConfig) (*Core, error) {
 		})
 	}
 	c.seal.SetCore(c)
+	return c, nil
+}
 
-	if err := coreInit(c, conf); err != nil {
+// NewCore is used to construct a new core
+func NewCore(conf *CoreConfig) (*Core, error) {
+	var err error
+	c, err := NewCoreUninit(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = coreInit(c, conf); err != nil {
```

Comment:
If we don't init the core object, we can't call the helper that checks if the autoloader license exists.

However, `VAULT_LICENSE` should contain the license if not empty, and `VAULT_LICENSE_PATH` should contain a filesystem license path to a license file (regardless of the state vault is in). 

There's also a field on the hcl config called `license_path` which should also contain the license path (same as env var able).

One of those three fields have to have something in it, otherwise autoloading cannot work.